crates/ruff_linter/src/message/gitlab.rs

```diff
@@ -61,23 +62,20 @@ impl Serialize for SerializedMessages<'_> {
         let mut fingerprints = HashSet::<u64>::with_capacity(self.diagnostics.len());
 
         for diagnostic in self.diagnostics {
-            let start_location = diagnostic.expect_ruff_start_location();
-            let end_location = diagnostic.expect_ruff_end_location();
-
             let filename = diagnostic.expect_ruff_filename();
-            let lines = if self.context.is_notebook(&filename) {
+
+            let start_location;
+            let end_location;
+
+            if self.context.is_notebook(&filename) {
                 // We can't give a reasonable location for the structured formats,
                 // so we show one that's clearly a fallback
-                json!({
-                    "begin": 1,
-                    "end": 1
-                })
+                start_location = LineColumn::default();
+                end_location = LineColumn::default();
             } else {
-                json!({
-                    "begin": start_location.line,
-                    "end": end_location.line
-                })
-            };
+                start_location = diagnostic.expect_ruff_start_location();
+                end_location = diagnostic.expect_ruff_end_location();
+            }
```

Comment:
Nit

```rust
let (start_location, end_location) = if self.context.is_notebook(&filename) {
	Default::default()
} else {
	start_location = diagnostic.expect_ruff_start_location();
  end_location = diagnostic.expect_ruff_end_location();
};
```

Reply:
Did you mean

```rust
            let (start_location, end_location) = if self.context.is_notebook(&filename) {
                // We can't give a reasonable location for the structured formats,
                // so we show one that's clearly a fallback
                (LineColumn::default(), LineColumn::default())
            } else {
                (
                    diagnostic.expect_ruff_start_location(),
                    diagnostic.expect_ruff_end_location(),
                )
            };
```

Reply:
Oh yes, sorry. The else branch should return a tuple. I think using `Default::default` might work for the notebook branch (I think)